packages/suite/src/views/dashboard/components/PortfolioCard/components/Exception.tsx

```diff
@@ -91,6 +102,7 @@ const Container = ({ title, description, cta, dataTestBase }: ContainerProps) =>
                         isLoading={isLocked()}
                         onClick={a.action}
                         data-testid={`@exception/${dataTestBase}/${a.variant || 'primary'}-button`}
+                        ref={retryButtonRef}
```

Comment:
I would expect that using a ref in a map can lead to unintended consequences.

I think you need something like
```
ref={a.label === 'TR_RETRY' ? retryButtonRef : null}  // assign ref conditionally
```

but tbh is it even needed? you assign the action you need in this component, you can "click" on it using this action `onClick={a.action}`

This is probably not a good approach either but what you wanted to do can be simply done by this
```
    useEffect(() => {
        if (isOnline) {
            const retryAction = actions.find(a => a.label === 'TR_RETRY');
            if (retryAction) {
                retryAction.action();
            }
        }
    }, [isOnline, actions]);
```

Have you considered using redux middleware? modifying this component like this can again lead to unintended consequences.

Reply:
You're right, thank you.

Reply:
> Have you considered using redux middleware? modifying this component like this can again lead to unintended consequences

?

Reply:
Sorry, I haven't considered using it, I just did a little research, and just to clarify: I'd create a redux middleware function that would watch a connection status and refresh anytime the device goes offline -> online, right? So it wouldn't even come to the moment when the user would be able to refresh the page manually and the button would be obsolete?